pandas/core/frame.py

```diff
@@ -8000,7 +8000,7 @@ def update(
             if mask.all():
                 continue
 
-            self[col] = expressions.where(mask, this, that)
+            self.loc[:, col] = expressions.where(mask, this, that)
```

Comment:
Since this writes into the underlying array, this is a change in behavior. Not sure if this is desireable.

Reply:
@phofl  Thanks for replying. per docstring under `update` that's what it's doing. 
`Modify in place using non-NA values from another DataFrame. Aligns on indices. There is no return value.`
it's just not doing in an efficient manner, since after 1.4.0 the setitem is being used for assignment, this needs to be optimized.  
do you mind tag other core contribute to take a look at this.
I really think this should be the way doing it.

Reply:
@phofl is right
this is an api change and cannot be back ported
nor is the right way

Reply:
repeated updates are not idiomatic

Reply:
@jreback  thanks for replying, I don't want to just jump in and change the API fundamentally but I want to understand you. the thing I am trying to fix is to speed up the operation. but didn't mean to have other impacts. 
to me, `df[col] = "a" ` vs `df.loc[:, col]="a"`  doesn't have much difference but the latter is much faster. this is only thing I meant to change. as regards to `repeated updates are not idiomatic`, i think this is what the existing  code is doing(in a for loop), and since docstring says it is to update it in place, isn't the change just right on its purpose? but the `df[col] = "a" ` is doing copy under the hood and it also slow it down.